src/Microsoft.Health.Dicom.Core/Microsoft.Health.Dicom.Core.csproj

```diff
@@ -8,6 +8,7 @@
   <ItemGroup>
     <PackageReference Include="Azure.Core" />
     <PackageReference Include="Azure.Identity" />
+    <PackageReference Include="Azure.Storage.Blobs" />
```

Comment:
There shouldn't be a blob dependency here. Perhaps there should be an exception that implements `DataStoreRequestFailedException` with blob-specific logic

Reply:
Ended up splitting the exception extentions into `BlobStorageErrorExtensions` and `AzureStorageErrorExtensions`, since the method that `DataStoreRequestFailedException` needs is not blob specific